src/app/WriteClient.h

```diff
@@ -281,9 +334,88 @@ class WriteClient : public Messaging::ExchangeDelegate
         return CHIP_NO_ERROR;
     }
 
+    CHIP_ERROR EnsureListStarted(const ConcreteDataAttributePath & attributePath)
```

Comment:
does this need to be inline? Could we move move things to the cpp file to make the header easier to read if people want to see the API?